### DISCUSSION TITLE
Two encoder/submit calls

### DISCUSSION THREAD

**Discussion by jaankaup** [gfx-rs/wgpu#5743] [General]
When I call two consecutive encoder/submit calls with drawing commands the whole program freezes. I have tested this with my laptop (rtx3060) and desktop machine (rtx 4070 ti). On windows however there is no problem at all. Does anyone have any idea what could be causing this.

The most simple example of this problems is to duplicate hello triangel code:

                        let frame = surface
                            .get_current_texture()
                            .expect("Failed to acquire next swap chain texture");
                        let view = frame
                            .texture
                            .create_view(&wgpu::TextureViewDescriptor::default());

                        // The first encoder/submit pair
                        let mut encoder =
                            device.create_command_encoder(&wgpu::CommandEncoderDescriptor {
                                label: None,
                            });
                        {
                            let mut rpass =
                                encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
                                    label: None,
                                    color_attachments: &[Some(wgpu::RenderPassColorAttachment {
                                        view: &view,
                                        resolve_target: None,
                                        ops: wgpu::Operations {
                                            load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
                                            store: wgpu::StoreOp::Store,
                                        },
                                    })],
                                    depth_stencil_attachment: None,
                                    timestamp_writes: None,
                                    occlusion_query_set: None,
                                });
                            rpass.set_pipeline(&render_pipeline);
                            rpass.draw(0..3, 0..1);
                        }
                        queue.submit(Some(encoder.finish()));

                        // The second encoder/submit pair (duplicate encoder/submit).
                        let mut encoder =
                            device.create_command_encoder(&wgpu::CommandEncoderDescriptor {
                                label: None,
                            });
                        {
                            let mut rpass =
                                encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
                                    label: None,
                                    color_attachments: &[Some(wgpu::RenderPassColorAttachment {
                                        view: &view,
                                        resolve_target: None,
                                        ops: wgpu::Operations {
                                            load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
                                            store: wgpu::StoreOp::Store,
                                        },
                                    })],
                                    depth_stencil_attachment: None,
                                    timestamp_writes: None,
                                    occlusion_query_set: None,
                                });
                            rpass.set_pipeline(&render_pipeline);
                            rpass.draw(0..3, 0..1);
                        }

                        queue.submit(Some(encoder.finish()));
                        frame.present();

This code makes gpu stuck on ubuntu/nvidia, but not on windows nvidia.

**Comment by SludgePhD** [comment]
This might be caused by https://github.com/gfx-rs/wgpu/issues/5693

**Reply by jaankaup** [reply]
This actually fixes it! Thanks!
